frontend/spec/controllers/spree/checkout_controller_spec.rb

```diff
@@ -170,10 +170,59 @@ def spree_post_address
         end
       end
 
+      # This is the only time that we need the 'set_payment_parameters_amount'
+      # controller code, because otherwise the transition to 'confirm' will
+      # trigger the 'add_store_credit_payments' transition code which will do
+      # the same thing here.
+      # Perhaps we can just remove 'set_payment_parameters_amount' entirely at
+      # some point?
+      context "when there is a checkout step between payment and confirm" do
+        before do
+          @old_checkout_flow = Spree::Order.checkout_flow
+          Spree::Order.class_eval do
+            insert_checkout_step :new_step, after: :payment
+          end
+        end
+
+        after do
+          Spree::Order.checkout_flow(&@old_checkout_flow)
+        end
+
+        let(:order) { create(:order_with_line_items) }
```

Comment:
Well, the point of extracting logic into another class is that you are building an interface, on which you can start *relying*. 

This test basically says, I do not rely on the `OrderUpdateAttributes`, so I will double check that everything is as expected there.

Other tests also double check, all that creates trouble .. slow tests, single error -> multiple test failures, etc. This talk [J.B. Rainsberger - Integrated Tests Are A Scam](https://vimeo.com/80533536) describes the problem well.

So, maybe to do an `expect(OrderUpdateAttributes).to receive(:new).with(...).and_return(MockOrderUpdater)` and then do an expectation for `#apply` on the `MockOrderUpdater`.

I understand that this #update method is very complicated, and maybe in this case it is better to leave it as it is, but wanted to point out the pattern for not so "do it all" methods.